source/views/components/filter/filter-toolbar-button.js

```diff
@@ -0,0 +1,125 @@
+// @flow
+import * as React from 'react'
+import {StyleSheet, Text, Platform, TouchableOpacity} from 'react-native'
+import Icon from 'react-native-vector-icons/Ionicons'
+import type {FilterType} from './types'
+import {FilterPopover} from './filter-popover'
+import * as c from '../colors'
+
+const buttonStyles = StyleSheet.create({
+	button: {
+		flexDirection: 'row',
+		alignItems: 'center',
+		marginRight: 10,
+		paddingHorizontal: 8,
+		paddingVertical: 5,
+		marginVertical: 8,
+		borderWidth: 1,
+		borderRadius: 2,
+	},
+	activeButton: {
+		backgroundColor: c.toolbarButtonBackground,
+		borderColor: c.toolbarButtonBackground,
+	},
+	inactiveButton: {
+		borderColor: c.iosDisabledText,
+	},
+	activeText: {
+		color: c.toolbarButtonForeground,
+	},
+	inactiveText: {
+		color: c.iosDisabledText,
+	},
+	text: {
+		fontSize: 16,
+	},
+	textWithIcon: {
+		paddingRight: 8,
+	},
+})
+
+type Props = {
+	filter: FilterType,
+	isActive: boolean,
+	onPopoverDismiss: (filter: FilterType) => any,
+	style?: any,
+	title: string,
+}
+
+type State = {
+	popoverVisible: boolean,
+}
+
+export class FilterToolbarButton extends React.PureComponent<Props, State> {
+	touchable: TouchableOpacity
+
+	state = {
+		popoverVisible: false,
+	}
+
+	openPopover = () => {
+		this.setState(() => ({popoverVisible: true}))
+	}
+
+	onClosePopover = (filter: FilterType) => {
+		this.props.onPopoverDismiss(filter)
+		this.setState(() => ({popoverVisible: false}))
+	}
+
+	render() {
+		const {filter, isActive, style, title} = this.props
+		const {popoverVisible} = this.state
+		const icon = Platform.select({
+			ios: 'ios-arrow-down',
+			android: 'md-arrow-dropdown',
+		})
+
+		let activeButtonStyle = isActive
+			? buttonStyles.activeButton
+			: buttonStyles.inactiveButton
+		let activeContentStyle = isActive
+			? buttonStyles.activeText
+			: buttonStyles.inactiveText
+
+		let textWithIconStyle = icon ? buttonStyles.textWithIcon : null
+		let activeTextStyle = {
+			fontWeight: isActive && Platform.OS === 'android' ? 'bold' : 'normal',
+		}
+
+		if (filter.type === 'list') {
+			if (!filter.spec.options.length) {
+				return null
+			}
+		}
+
+		return (
+			<React.Fragment>
+				<TouchableOpacity
```

Comment:
Only more question: why is this a TouchableOpacity instead of a Touchable? Touchable automatically switches to use "ripples" on android, which is usually nice, and is why I'm asking.

Reply:
no reason!

Reply:
Yeah, sorry. You won't be able to just switch it for a ref= on Touchable; you'll have to forward the ref through Touchable to TouchableOpacity/Highlight/NativeFeedback (when you see innerRef= or getRef=, it's forwarding the ref.)

I can give you an example if you would like.

Reply:
Or, we can file this under "things Hawken should look at after it's merged" because I just realized that I don't know how I want you to handle typing that three-way intersection.

Reply:
See 9fdadea173e1d3f872a64004d72454533d1df45f for this.